frontend/packages/process-editor/src/hooks/useBpmnViewer.ts

```diff
@@ -1,16 +1,22 @@
-import { MutableRefObject, useRef, useEffect } from 'react';
+import { MutableRefObject, useRef, useEffect, useState } from 'react';
```

Comment:
I think this suggestion is easier to explain with pair-programming, but I think we can simplify the error handling by do something like this:

```typescript
import { MutableRefObject, useRef, useEffect, useState } from 'react';
import BpmnJS from 'bpmn-js/dist/bpmn-navigated-viewer.development.js';
import { useBpmnContext } from '../contexts/BpmnContext';
import { useTranslation } from 'react-i18next';

// Wrapper around bpmn-js to Reactify it

type BpmnViewerError = 'noDiagram' | 'noProcess' | 'unknown';

const bpmnViewerErrorMap: Record<string, BpmnViewerError> = {
  'no diagram to display': 'noDiagram',
  'no process or collaboration to display': 'noProcess',
};

type UseBpmnViewerResult = {
  canvasRef: MutableRefObject<HTMLDivElement>;
  bpmnViewerError: BpmnViewerError | undefined;
};

export const useBpmnViewer = (): UseBpmnViewerResult => {
  const { t } = useTranslation();
  const { bpmnXml } = useBpmnContext();
  const canvasRef = useRef<HTMLDivElement | null>(null);
  const [bpmnViewerError, setBpmnViewerError] = useState<BpmnViewerError | undefined>(undefined);

  useEffect(() => {
    if (!canvasRef.current) {
      console.log('Canvas reference is not yet available in the DOM.');
      return;
    }

    const viewer = new BpmnJS({ container: canvasRef.current });

    const initializeViewer = async () => {
      try {
        await viewer.importXML(bpmnXml);
      } catch (exception) {
        setBpmnViewerError(bpmnViewerErrorMap[exception.message] || 'unknown');
      }
    };

    initializeViewer();
  }, [bpmnXml, t]);

  return { canvasRef, bpmnViewerError };
};
```